src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -1,10 +1,53 @@
-import React, { useCallback, useRef, useEffect } from 'react';
+import React, { Ref, useCallback, useRef, useEffect } from 'react';
 import { Vector3 } from 'three';
 import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
 import { clamp } from 'lodash-es';
+import { useMeasure } from 'react-use';
+import { useHeatmapState } from './store';
 
 const ZOOM_FACTOR = 0.95;
 
+export function useHeatmapSize<T>(
+  dims: [number, number],
+  axisOffset: [number, number]
+): [(elem: T) => void, { width: number; height: number } | undefined] {
+  const { hasOrthoCoords } = useHeatmapState();
```

Comment:
I find `hasOrthoCoords` is not that clear. What about `keepImageAspectRatio` or `keepAspectRatio` ?